server/cleanup.ts

```diff
@@ -19,3 +24,68 @@ async function cleanupPostgres() {
   );
   console.timeEnd('[CLEANUP]');
 }
+
+function cleanupRoles() {
+  console.time('[DISCORD CLEANUP]');
+  const client = new DiscordBot({
+    intents: [Intents.FLAGS.GUILDS, Intents.FLAGS.GUILD_MEMBERS],
+  });
+  client.once('ready', async () => {
+    if (!postgres) {
+      return;
+    }
+    const guild = client.guilds.cache.get(config.DISCORD_SERVER_ID);
+    const members = await guild?.members.fetch();
+    // get username and discriminator of all users who
+    // currently have a sub role assigned on discord
+    const membersWithRole = members
+      ?.filter((member) =>
+        member.roles.cache.some(
+          (role) => role.id === config.DISCORD_SUB_ROLE_ID
+        )
+      )
+      .map((member) => ({
+        username: member.user.username,
+        discriminator: member.user.discriminator,
+      }));
+    const subs = await getAllActiveSubscriptions();
+
+    try {
+      // get all known users who have been assigned a sub role
+      const result = await postgres?.query('SELECT * FROM discord');
```

Comment:
I'm fine with this being one-way (e.g. we don't remove roles for former subscribers).